kafka-handson/kafka-handson.md

```diff
@@ -0,0 +1,401 @@
+# Kafka Hands-On
+
+* [Deploy the Operator](#deploy-the-operator)
+* [Deploy Kafka](#deploy-kafka)
+* [Scale up](#scale-up)
+* [Configure Node-affinity](#node-affinity)
+* [Run a Producer and Consumer](#run-a-producer-and-consumer)
+* [Describe a Topic](#describe-a-topic)
+* [Deploy Connect and a Connector](#deploy-connect-and-a-connector)
+* [Deploy Control Center](#deploy-control-center)
+* [Deploy Schema Registry](#deploy-the-schema-registry)
+
+In this hands-on, we will deploy Confluent for Kubernetes, which is a Kubernetes native Kafka operator. For more information, go to the official documentation: https://docs.confluent.io/operator/current/overview.html
+
+
+## Deploy the Operator
+
+First, we will create a namespace in which the Kafka Platform should run.
+
+Create a namespace called "operator".
+
+<details>
+  <summary>Solution</summary>
+
+```bash
+kubectl create namespace operator
+```
+
+</details>
+
+Then we will deploy the operator.
+
+```bash
+helm upgrade --install --values operator-values.yaml --namespace operator operator confluent-for-kubernetes-2.1.0/helm/confluent-for-kubernetes
+```
+
+Check if the operator pod is running.
+
+<details>
+  <summary>Solution</summary>
+
+```bash
+kubectl get pods -n operator
+```
+
+You should see a pod running similar to this one:
+
+```
+NAME                                  READY   STATUS    RESTARTS   AGE
+confluent-operator-5699dd58f7-sfgkx   1/1     Running   0          37s
+```
+
+You can also see your operator pod in the Kubernetes Dashboard:
+
+http://localhost:8001/api/v1/namespaces/kubernetes-dashboard/services/https:kubernetes-dashboard:/proxy/#/pod?namespace=operator
+
+</details>
+
+## Deploy Kafka
+
+Now that the operator is running, it will react if a new custom resource file is applied to the cluster and will create or update resources as declared.
+
+Install Zookeeper:
+
+```bash
+kubectl apply -f zookeeper.yaml -n operator
+```
+
+<details>
+  <summary>Solution</summary>
+
+Verify that the zookeeper pod is running:
```

Comment:
Should be outside the solution block. Otherwise very difficult to figure out what you need to do without opening the solution.